### PR TITLE
SW-5122 Make the welcome banner dismissible

### PR DESCRIPTION
**Pull request by nickgraz**
- Make `DismissableWrapper` which makes it easy to embed components which can be dismissed according to their user preferences
- Wrap the `WelcomeBanner`

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2518?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2518** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2518?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2516** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2516?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2515** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2515?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2514** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2514?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2513** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2513?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nickgraz and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/components/common/DismissibleWrapper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { useMemo, useState } from 'react';
+
+import { useUser } from 'src/providers';
+
+export type DismissibleWrapperProps = {
+  dontShowAgainPreferenceName: string;
+  children: (onClose: () => void) => JSX.Element;
+};
+
+export default function DismissibleWrapper(props: DismissibleWrapperProps): JSX.Element | null {
```
Comment (karthikbtf) [inline src/components/common/DismissibleWrapper/index.tsx:21]: Nice

**Review by karthikbtf — APPROVED** [review]
Nice wrapper, wondering if we have any existing use-cases to update.

**Comment by nickgraz** [comment]
I yoinked this from your planting sites code so I can update that once this is merged in.

**Comment by joshbuchea** [comment]
Nice wrapper!

**Comment by nickgraz** [comment]
### Merge activity

* **Apr 16, 7:05 PM EDT**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2518).

**Pull request merged**